### PR TITLE
:bricks:Conditional block redesign

### PR DESCRIPTION
**Pull request by Dancunrandy**
# Description

Re-designed conditional block
Fixes #722 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Screenshot (optional)
Screenshot from 2023-11-05 12-14-05
Screenshot from 2023-11-16 22-57-45
Screenshot from 2023-11-16 22-58-16

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**: (optional)
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Dancunrandy** [comment]
@LemmyMwaura , resolved the checkbox, it's now clickable

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/providers/conditional-block.jsplumb.ts [hunk 1]
```diff
@@ -23,10 +23,21 @@ export function _ConditionalBlockDecoratePlumb(block: ConditionalBlock, comp: Co
     cssClass:"block_endpoint",
 
     endpoint: 'Dot',
-    anchor: "Left",
+    anchor: [0,0.17,0,1],
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/main/src/lib/providers/conditional-block.jsplumb.ts:26]: Leave this as 'Left' for now

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/providers/conditional-block.jsplumb.ts [hunk 2]
```diff
@@ -23,10 +23,21 @@ export function _ConditionalBlockDecoratePlumb(block: ConditionalBlock, comp: Co
...
+    // Whether the anchor is target (Other Block -> This Block)
+    target: true,
+    cssClass:"block_endpoint",
+
+    endpoint: 'Dot',
+    anchor: [1,0.9,0,1],
+    maxConnections: -1
+
+  });
+
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/main/src/lib/providers/conditional-block.jsplumb.ts:40]: remove this, 'the second endpoint'

File: libs/features/convs-mgr/stories/blocks/library/conditional-block/src/lib/component/conditional-block/conditional-block.component.html [hunk 3]
```diff
@@ -1,44 +1,33 @@
...
           <button (click)="deleteInput(i)" class="listgone"><i class="far fa-trash-alt"></i></button>
-          <app-option-input-field [jsPlumb]="jsPlumb" [blockFormGroup]="conditionalBlockForm" [formGroupNameInput]="i"
-            class="input" [charMaxlength]="listOptionInputLimit"></app-option-input-field>
+          <app-option-input-field [jsPlumb]="jsPlumb" [blockFormGroup]="conditionalBlockForm" [formGroupNameInput]="i" class="input" [charMaxlength]="listOptionInputLimit"></app-option-input-field>
         </div>
       </div>
     </div>
-
-    <app-default-option-field [jsPlumb]="jsPlumb" [blockFormGroup]="conditionalBlockForm"></app-default-option-field>
-
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/conditional-block/src/lib/component/conditional-block/conditional-block.component.html:38]: return the default option for now. (we'll remove it later)

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]

**Comment by LemmyMwaura** [comment]
Also, kindly resolve your conflicts

**Comment by Dancunrandy** [comment]
@LemmyMwaura resolved the conflicts, activated default and restored previous dot position

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/utils/transformers/conditional-block.jsplumb.ts [hunk 4]
```diff
@@ -27,6 +27,17 @@ export function _ConditionalBlockDecoratePlumb(block: ConditionalBlock, comp: Co
...
+  // jsPlumb.addEndpoint(comp.location.nativeElement, {
+  //   // Whether the anchor is target (Other Block -> This Block)
+  //   target: true,
+  //   cssClass:"block_endpoint",
+
+  //   endpoint: 'Dot',
+  //   anchor: [1,0.9,0,1],
+  //   maxConnections: -1
+
+  // });
```
[the remark this answers is not in the document]
Comment (Dancunrandy) [reply libs/features/convs-mgr/stories/blocks/library/main/src/lib/utils/transformers/conditional-block.jsplumb.ts]: @have done the synching and removed completely the above part as requested.

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
Just noticed the add option is not properly styled. Can you take care of this please? Also, the select option should be disabled when the custom variable checkbox is selected.

Screenshot 2023-11-16 at 12 39 02

**Comment by Dancunrandy** [comment]
@LemmyMwaura  styled the add option properly and the select option is disabled when the custom variable checkbox is selected.

File: libs/features/convs-mgr/stories/blocks/library/conditional-block/src/lib/component/conditional-block/conditional-block.component.ts [hunk 5]
```diff
@@ -24,22 +24,20 @@ export class ConditionalBlockComponent<T> implements OnInit, AfterViewInit, OnDe
...
   vars$: Observable<string[]>;
 
-  private _sBs = new SubSink();
   readonly listOptionInputLimit = 20;
   readonly listOptionsArrayLimit = 10;
 
-  constructor(private _fb: FormBuilder, 
-              private variables: VariablesService) 
-  {
+  constructor(private fb: FormBuilder, private variables: VariablesService) {
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/conditional-block/src/lib/component/conditional-block/conditional-block.component.ts:35]: Is there any reason this was changed from _fb to fb or _sBs to subsink ??

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
If not kindly rename but everything else looks good to me.

File: libs/features/convs-mgr/stories/blocks/library/conditional-block/src/lib/component/conditional-block/conditional-block.component.ts (hunk 5 above)
Comment (Dancunrandy) [reply libs/features/convs-mgr/stories/blocks/library/conditional-block/src/lib/component/conditional-block/conditional-block.component.ts:35]: No, there wasn't, i think my vs code  did auto correction while fixing the errors, i can restore the original code if you like.

Comment (Dancunrandy) [reply libs/features/convs-mgr/stories/blocks/library/conditional-block/src/lib/component/conditional-block/conditional-block.component.ts:35]: ok, i will

**Comment by Dancunrandy** [comment]
@LemmyMwaura ,have renamed the code

**Comment by LemmyMwaura** [comment]
Kindly sync the latest changes, the design is the only missing thing. I refactored the design for the question, list and event blocks etc. You can just copy the design? including the options which are still in the old design format. @Dancunrandy

**Review by LemmyMwaura — APPROVED** [review]

**Pull request merged**